### PR TITLE
Upgrade some dependencies

### PR DESCRIPTION
**Pull request by wb9688**
#### What is it?
- [ ] Bug fix (user facing)
- [ ] Feature (user facing)
- [x] Code base improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Agreement
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

### CODE REVIEW

**Comment by Stypox** [comment]
Have you tested on 4.4?

**Comment by wb9688** [comment]
@Stypox: Not thoroughly, but I was going to.

**Comment by wb9688** [comment]
Reading licenses is broken

File: app/src/main/java/org/schabi/newpipe/about/LicenseFragmentHelper.java [hunk 1]
```diff
@@ -26,8 +28,8 @@ public LicenseFragmentHelper(@Nullable final Activity activity) {
         weakReference = new WeakReference<>(activity);
     }
 
-    private static String getFinishString(final Activity activity) {
-        return activity.getApplicationContext().getResources().getString(R.string.finish);
+    private static String getFinishString(final Context context) {
+        return context.getString(R.string.finish);
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/about/LicenseFragmentHelper.java:32]: Nothing to do with dependency updates, but what's the purpose of this function? `context.getString(R.string.finish)` is already explicative, readable and not long, so why encapsulate it in a function?

Comment (wb9688) [reply app/src/main/java/org/schabi/newpipe/about/LicenseFragmentHelper.java:32]: I was wondering that as well… I'll fix it later today

**Comment by TobiGr** [comment]
> is there a protocol that can be followed to fully test such app-wide changes?

@Stypox Test _everything_ on KitKat, Lollipop, latest and a version (maybe Oreo) in between. With so many upgraded dependencies almost everything is somehow affected by this change.

**Comment by wb9688** [comment]
@TobiGr: I think he meant more like what is _"everything"_? NewPipe has so many features that you could easily forget one.

File: app/build.gradle [hunk 2]
```diff
@@ -79,22 +79,23 @@ android {
...
+    icepickVersion = '3.2.0'
+    checkstyleVersion = '8.32'
+    stethoVersion = '1.5.1'
+    leakCanaryVersion = '2.2'
+    exoPlayerVersion = '2.10.8'
+    androidxLegacyVersion = '1.0.0'
+    androidxLifecycleVersion = '2.2.0'
+    androidxRoomVersion = '2.2.5'
+    groupieVersion = '2.8.0'
+    markwonVersion = '4.3.1'
```
Comment (B0pol) [inline app/build.gradle:91]: Why did you change the order? And if we change the order, it should be for lexicographical order.

Comment (wb9688) [reply app/build.gradle:91]: I changed the order to be the same as down there. Down there I changed the order in a way that seems more logical to me.

**Comment by Stypox** [comment]
Btw I also tested on my 7.0 phone and everything works fine

**Comment by B0pol** [comment]
Tested on 9.0 phone and Android TV 7.1.2 and I haven't found an issue.

**Comment by wb9688** [comment]
I tested it on AVD API 19 and on my phone with Android 10. I'll rebase and do what @Stypox suggested.

File: app/src/main/java/org/schabi/newpipe/about/LicenseFragmentHelper.java [hunk 3]
```diff
@@ -26,28 +28,24 @@ public LicenseFragmentHelper(@Nullable final Activity activity) {
...
-    public static String getFormattedLicense(final Context context, final License license) {
+    private static String getFormattedLicense(final Context context, final License license) {
         if (context == null) {
-            throw new NullPointerException("context is null");
+            throw new IllegalArgumentException("context is null");
         }
         if (license == null) {
-            throw new NullPointerException("license is null");
+            throw new IllegalArgumentException("license is null");
         }
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/about/LicenseFragmentHelper.java:43]: This is also unrelated to dependency updates, but this block of code should be replaced by `@NotNull` parameters, or just removed alltoghether, because a NullPointerException would be thrown anyway crashing the app, so it makes no sense to check for nullity and then do not do anything different.

**Comment by TobiGr** [comment]
On KitKat main page tabs are not loaded again when new tabs are added through the settings.
E.g. I added the history tab and moved it to the top of the list. Instead of seeing the history when going back, I see trending content.

File: app/src/main/java/org/schabi/newpipe/about/LicenseFragmentHelper.java (hunk 3 above)
Comment (wb9688) [reply app/src/main/java/org/schabi/newpipe/about/LicenseFragmentHelper.java:43]: That's why I made it an `IllegalArgumentException`, though making it `@NotNull` is probably better indeed

**Comment by wb9688** [comment]
@TobiGr: I just rebased, but I can't reproduce your bug.

**Comment by Stypox** [comment]
@TobiGr I think this has to do with #3312 and is unrelated from this PR.

File: app/src/debug/java/org/schabi/newpipe/DebugApp.kt [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    override fun attachBaseContext(base: Context) {
+        super.attachBaseContext(base)
+        MultiDex.install(this)
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        initStetho()
+
+        AppWatcher.config = AppWatcher.config.copy(watchDurationMillis = 10000)
```
Comment (TobiGr) [inline app/src/debug/java/org/schabi/newpipe/DebugApp.kt:23]: Please (re-)add the comment about the delay. The new naming is less understandable.

File: app/build.gradle [hunk 5]
```diff
@@ -163,10 +163,10 @@ dependencies {
         exclude module: 'support-annotations'
     }
 
-    implementation 'com.github.TeamNewPipe:NewPipeExtractor:f3913e241e379adf0091319091e8f895c5fcfd07'
+    implementation 'com.github.wb9688:NewPipeExtractor:a28cf8a26cf1dce8b46b1be049b00265aa971189'
```
Comment (TobiGr) [inline app/build.gradle:166]: We should not forget to change this line once the extractor PR is merged.
